test/mix/tasks/docs_test.exs

```diff
@@ -181,4 +181,8 @@ defmodule Mix.Tasks.DocsTest do
       ] = run([], app: :umbrella, apps_path: "apps/", docs: [ignore_apps: [:foo]])
     end)
   end
+
+  test "accepts warnings_as_errors in :warnings_as_errors" do
+    assert catch_exit(run([], app: :ex_doc, docs: [warnings_as_errors: true])) == {:shutdown, 1}
```

Comment:
🤔 Individually, this test should fail, it's assuming that the state of the `WarningCounter` is greater than zero when is not the case.